test/AutoDiff/sil_differentiability_witness_silgen.swift

```diff
@@ -0,0 +1,54 @@
+// RUN: %target-swift-frontend -emit-silgen %s | %target-sil-opt | %FileCheck %s
+
+// Test SIL differentiability witness SIL generation.
```

Comment:
Todo: add more tests, namely `@differentiable`/`@differentiating` attributes for different original declaration kinds (initializers, subscripts, methods, computed properties).

Reply:
Added some more tests.